### PR TITLE
docs: fix `replacementNameTemplate` samples

### PR DESCRIPTION
**Pull request by viceice**
## Changes
- found while testing new docs

## Context
- #21076

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/configuration-options.md [hunk 1]
```diff
@@ -2231,7 +2231,7 @@ For example, the following package rule can be used to replace the registry for
   "packageRules": [
     {
       "matchDatasources": ["docker"],
-      "matchPackagePatterns": ["^docker.io/.*)"],
+      "matchPackagePatterns": ["^docker.io/.*"],
```
Comment (viceice) [inline docs/usage/configuration-options.md:2234]:
```suggestion
      "matchPackagePatterns": ["^docker.io/.+"],
```

File: docs/usage/configuration-options.md [hunk 2]
```diff
@@ -2231,7 +2231,7 @@ For example, the following package rule can be used to replace the registry for
   "packageRules": [
     {
       "matchDatasources": ["docker"],
-      "matchPackagePatterns": ["^docker.io/.*)"],
+      "matchPackagePatterns": ["^docker.io/.+"],
```
Comment (viceice) [inline docs/usage/configuration-options.md:2234]:
```suggestion
      "matchPackagePatterns": ["^docker\\.io/.+"],
```

File: docs/usage/configuration-options.md [hunk 3]
```diff
@@ -2231,7 +2231,7 @@ For example, the following package rule can be used to replace the registry for
...
       "replacementNameTemplate": "{{{replace 'docker.io/' 'ghcr.io/' packageName}}}"
```
Comment (viceice) [inline docs/usage/configuration-options.md:2235]:
```suggestion
      "replacementNameTemplate": "{{{replace 'docker\\.io/' 'ghcr.io/' packageName}}}"
```

File: docs/usage/configuration-options.md [hunk 4]
```diff
@@ -2246,13 +2246,13 @@ Or, to add a registry prefix to any `docker` images that do not contain an expli
     {
       "description": "official images",
       "matchDatasources": ["docker"],
-      "matchPackagePatterns": ["^[a-z-]+(:|$).*"],
+      "matchPackagePatterns": ["^[a-z-]+"],
```
Comment (viceice) [inline docs/usage/configuration-options.md:2249]:
```suggestion
      "matchPackagePatterns": ["^[a-z-]+$"],
```

File: docs/usage/configuration-options.md [hunk 5]
```diff
@@ -2246,13 +2246,13 @@ Or, to add a registry prefix to any `docker` images that do not contain an expli
...
       "replacementNameTemplate": "some.registry.org/library/{{{packageName}}}"
     },
     {
       "description": "non-official images",
       "matchDatasources": ["docker"],
-      "matchPackagePatterns": ["^[a-z-]+\\/[a-z-]+(:|$).*"],
+      "matchPackagePatterns": ["^[a-z-]+\\/[a-z-]+"],
```
Comment (viceice) [inline docs/usage/configuration-options.md:2255]:
```suggestion
      "matchPackagePatterns": ["^[a-z-]+/[a-z-]+$"],
```

File: docs/usage/configuration-options.md [hunk 6]
```diff
@@ -2231,8 +2231,8 @@ For example, the following package rule can be used to replace the registry for
   "packageRules": [
     {
       "matchDatasources": ["docker"],
-      "matchPackagePatterns": ["^docker.io/.*)"],
-      "replacementNameTemplate": "{{{replace 'docker.io/' 'ghcr.io/' packageName}}}"
+      "matchPackagePatterns": ["^docker\\.io/.+"],
+      "replacementNameTemplate": "{{{replace 'docker\\.io/' 'ghcr.io/' packageName}}}"
```
Comment (setchy) [inline docs/usage/configuration-options.md:2235]: Ah interesting.... 

would using `matchPackagePrefixes` instead avoid needing to escape the `.`?

**Review by setchy — APPROVED** [review]

**Review by secustor — APPROVED** [review]

**Pull request merged**